amd_openvx_extensions/amd_rpp/source/tensor/NonLinearBlend.cpp

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct NonLinearBlendLocalData {
+    vxRppHandle *handle;
+    vx_uint32 deviceType;
+    RppPtr_t pSrc1;
+    RppPtr_t pSrc2;
+    RppPtr_t pDst;
+    Rpp32f *pStdDev;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    RpptROI *pSrcRoi;
+    RpptRoiType roiType;
+    vxTensorLayout inputLayout;
+    vxTensorLayout outputLayout;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+static vx_status VX_CALLBACK refreshNonLinearBlend(vx_node node, const vx_reference *parameters, vx_uint32 num, NonLinearBlendLocalData *data) {
+    vx_status status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[4], 0, data->pSrcDesc->n, sizeof(Rpp32f), data->pStdDev, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
```

Comment:
[Blocking] For NF layouts this descriptor's n equals batch*frames, but `pStdDev` is passed per sample. When F > 1 this tries to read n entries from the array, which either overruns or makes vxCopyArrayRange return VX_ERROR_INVALID_PARAMETERS. Could we copy only inputTensorDims[0] elements and rely on the NF expansion below to fan them out?